pandas/core/groupby.py

```diff
@@ -4437,20 +4437,15 @@ def _reorder_by_uniques(uniques, labels):
 
 
 def _groupby_indices(values):
+    """
+    return a dictionary of groups -> indices
+    """
+    if not is_categorical_dtype(values):
+        values = Categorical(values)
```

Comment:
I meant in the non Categorical case, if `Grouping.labels` is already populated, could skip factorizing again in the Categorical?